### PR TITLE
feat(dynamic-sampling): Handles updating `ProjectConfig` with uniform DS rule for v2 [TET-465]

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR forces your uniform rule by your plan
or respect old logic. If both feature flags are enabled
`dynamic-sampling-basic` flag takes the highest precedence.

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -200,6 +207,11 @@ def _get_project_config(project, full_config=True, project_keys=None):
 
             cfg["config"]["dynamicSampling"] = {"rules": active_rules}
 
+    # In this case we should override old conditionnal rules if they exists
+    # or just return uniform rule
+    if allow_dynamic_sampling_basic:
+        cfg["config"]["dynamicSampling"] = {"rules": [generate_uniform_rule(project)]}
```
Comment (andriisoldatenko) [inline src/sentry/relay/config/__init__.py:213]: do we want to capture_exception here?
E.g. if 
```
try:
    cfg["config"]["dynamicSampling"] = {"rules": [generate_uniform_rule(project)]}
except Exception:
        sentry_sdk.capture_exception()
```
Especially when `quotas.get_blended_sample_rate(project)` returns None?

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -200,6 +207,11 @@ def _get_project_config(project, full_config=True, project_keys=None):
 
             cfg["config"]["dynamicSampling"] = {"rules": active_rules}
 
+    # In this case we should override old conditionnal rules if they exists
+    # or just return uniform rule
+    if allow_dynamic_sampling_basic:
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:212]: Should this be an `elif`?

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/relay/config/__init__.py:213]: That's definitely the safest option. Though if `get_blended_sample_rate` returns `None`, there won't be an exception, will there? Best to handle that case separately.

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:212]: no we want to override previous rules, so like remove all condition rules on the fly and return only single uniform rule.

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:213]: yeah it's mostly question, cause we discussed wit @ahmedetefy to make it required and may be tweak function/ to raise an exception if it's None.

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (ahmedetefy) [reply src/sentry/relay/config/__init__.py:212]: Yep agreed here

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/relay/config/__init__.py:213]: I agree with @jjbayer here, I think we should handle that separately in `generate_uniform_rule` and just not return dynamic sampling config to relay

File: src/sentry/dynamic_sampling/utils.py [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 
+import sentry_sdk
+
 from sentry import quotas
 from sentry.models import Project
 
 UNIFORM_RULE_RESERVED_ID = 0
 
 
+class SampleRateIsNone(Exception):
```
Comment (ahmedetefy) [inline src/sentry/dynamic_sampling/utils.py:11]:
```suggestion
class NoneSampleRateException(Exception):
```

Comment (andriisoldatenko) [reply src/sentry/dynamic_sampling/utils.py:11]: fixed i n 9e32628d57

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (jjbayer) [reply src/sentry/relay/config/__init__.py:211]: Wouldn't it make sense to put this block _before_ line 187 then? No need to check the feature `organizations:server-side-sampling`, fetch the option `sentry:dynamic_sampling` and call `get_latest_release` if it's going to be overwritten anyway.

So I'm suggesting

```python
if allow_dynamic_sampling_basic:
   ...
elif features.has(
    "organizations:server-side-sampling",
    project.organization,
):
    ...

File: src/sentry/relay/config/__init__.py [hunk 4]
```diff
@@ -200,6 +206,16 @@ def _get_project_config(project, full_config=True, project_keys=None):
...
+        try:
+            cfg["config"]["dynamicSampling"] = {"rules": [generate_uniform_rule(project)]}
+        except NoneSampleRateException:
+            # just to be consistent with old code, where if there is no active active_rules
+            # we retrun empty list
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:216]:
```suggestion
            # we return empty list
```

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:211]: yeah good call, i'll fix.

**Review by jjbayer — APPROVED** [review]

**Review by ahmedetefy — APPROVED** [review]

**Pull request merged**